### PR TITLE
Download all progress 4811

### LINKED ISSUE #4811: Progress is not shown for Save All
1. Open an email with multiple attachments on Android 
2. Click on "save all"
3. See no indication of anything happening

We could show progressDialog even if it's slow. Ideally it would be non-blocking progress indicator.

## Test Notes
* [ ] It is best to test all of:
Chrome,
Firefox,
Safari,
Desktop Ubuntu and Windows,
iOS App,
Android App,
Chrome and Firefox on Android
and Safari on iOS
* [ ] iOS App won't show the "Save all" download option
* [ ] download a single file
  * [ ] make sure that the default download location is respected in case it is set (should be download folder on android) -> file appears in this folder
  * [ ] if the setting to ask for the destination before downloading is enabled in a browser, you should be prompted to choose a location
  * [ ] make sure that a spinning progress dialog is visible for the whole time downloading (even when selecting the downloaded files destination folder)
  * [ ] downloading without internet connection will give an error message
* [ ] download multiple files with the "Save all" option
  * [ ] make sure that the default download location is respected in case it is set (should be download folder on android) -> file appears in this folder (on Desktop and Browser the files should be saved as a single zip file)
  * [ ] if the setting to ask for the destination before downloading is enabled in a browser, you should be prompted to choose a location
  * [ ] make sure that a progress dialog (indicating the completeness) is visible for the whole time downloading (even when selecting the downloaded files destination folder)
  * [ ] downloading without internet connection will give an error message for each file, mentioning the file name

### Regressions
* [ ] Opening single files works on all above mentioned platforms (there shouldn't be such an option in browsers)

### PR DESCRIPTION
**Pull request by nokhub**
fix #4811

### CODE REVIEW

File: src/file/FileControllerNative.ts [hunk 1]
```diff
@@ -24,30 +25,62 @@ export class FileControllerNative implements FileController {
 	 * Temporary files are deleted afterwards in apps.
 	 */
 	async download(file: TutanotaFile) {
-		await guiDownload(this.doDownload(file))
+		await guiDownload(
+			isAndroidApp() || isDesktop()
+				? this.doDownload(
+						new Array(file),
```
Comment (charlag) [inline src/file/FileControllerNative.ts:31]: I was sooo confused by `new Array`, just do `[file]`

File: src/mail/view/MailViewerViewModel.ts [hunk 2]
```diff
@@ -909,7 +909,17 @@ export class MailViewerViewModel {
 	}
 
 	async downloadAll(): Promise<void> {
-		await this.fileController.downloadAll(this.getNonInlineAttachments())
+		try {
+			await this.fileController.downloadAll(this.getNonInlineAttachments())
+		} catch (e) {
+			if (e instanceof FileOpenError) {
+				console.warn("FileOpenError", e)
+				await Dialog.message("canNotOpenFileOnDevice_msg") // FIXME extra error message?
```
Comment (charlag) [inline src/mail/view/MailViewerViewModel.ts:917]: noting FIXME for later

File: src/file/FileControllerNative.ts [hunk 3]
```diff
@@ -24,30 +25,62 @@ export class FileControllerNative implements FileController {
...
+					console.log("failed to delete file", file.location, e)
+				}
 			}
-		} finally {
-			if (temporaryFile) {
+		}
+	}
+
+	private async doDownload<T>(
+		tutanotaFiles: TutanotaFile[],
```
Comment (charlag) [inline src/file/FileControllerNative.ts:62]: maybe we should make these arguments an object, it's easier to read this way

File: src/mail/view/MailViewerViewModel.ts (hunk 2 above)
Comment (nokhub) [reply src/mail/view/MailViewerViewModel.ts:917]: so are we going to keep it like this for now?

File: src/file/FileControllerNative.ts (hunk 1 above)
Comment (nokhub) [reply src/file/FileControllerNative.ts]: yea 🫠

File: src/file/FileControllerNative.ts (hunk 3 above)
Comment (nokhub) [reply src/file/FileControllerNative.ts]: that's true

**Pull request closed without merging**